cmd/push.go

```diff
@@ -53,10 +60,72 @@ func init() {
 
 type PushCmdParams struct {
 	AccountContextParams
-	ASU         string
-	allAccounts bool
-	force       bool
-	targeted    []string
+	ASU              string
+	sysAccUser       string // when present use
+	sysAcc           string
+	sysAccUserJwtOpt nats.Option
+	allAccounts      bool
+	force            bool
+	prune            bool
+	targeted         []string
+
+	accountList []string
+}
+
+func processResponse(report *store.Report, resp *nats.Msg) (bool, string, interface{}) {
+	serverResp := map[string]interface{}{}
+	if err := json.Unmarshal(resp.Data, &serverResp); err != nil {
+		report.AddError("failed to parse response: %v data: %s", err, string(resp.Data))
+	} else if srv, ok := serverResp["server"]; !ok {
+		report.AddError("server responded without server info: %s", string(resp.Data))
+	} else if srv, ok := srv.(map[string]interface{}); !ok {
+		report.AddError("server responded with bad server info: %s", string(resp.Data))
+	} else if error, hasErr := serverResp["error"]; hasErr {
+		if error, ok := error.(map[string]interface{}); ok {
+			report.AddError("server %s responded with error: %s", srv["name"], error["description"])
+		} else {
+			report.AddError("server %s responded with bad error: %v", srv["name"], error)
+		}
+	} else if data, ok := serverResp["data"]; !ok {
+		report.AddError("server %s responded without data: %s", srv["name"], string(resp.Data))
+	} else {
+		return true, srv["name"].(string), data
+	}
+	return false, "", nil
+}
+
+func systemAccountUser(ctx ActionCtx, sysAccUserName string) (string, string, nats.Option, error) {
+	if op, err := ctx.StoreCtx().Store.ReadOperatorClaim(); err != nil {
+		return "", "", nil, err
+	} else if accNames, err := friendlyNames(ctx.StoreCtx().Operator.Name); err != nil {
+		return "", "", nil, err
+	} else if sysAccName, ok := accNames[op.SystemAccount]; !ok {
```

Comment:
This is a similar thing to the problem of locating keys - we want to locate one. The algo should be what are the users for the system account, select one (unless a name for that user is specified in the command) otherwise any user will do. If the user doesn't have the permissions required, the request will fail and they can specify the name)

Reply:
of course any user for which have keys will do. 
However, I create the user sys by default now... so might as well use that one.

Reply:
then I would say, have a flag for the user, that defaults to that, then you solve both problems.